scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixOutput.java

```diff
@@ -0,0 +1,66 @@
+package scalafix.interfaces;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+public interface ScalafixOutput {
+
+    /**
+     * @return original path of the file to be fixed
+     */
+    Path getPath();
+
+    /**
+     *
+     * @return corresponding list of rules that has been evaluated
+     */
+    ScalafixRule[] getRules();
+
+    /**
+     *
+     * @return boolean true if there are no error when scalafix run on the file
+     */
+    boolean isSuccessful();
+
+    ScalafixError[] getErrors();
+
+    /**
+     *
+     * @return list of diagnostics
+     */
+    ScalafixDiagnostic[] getDiagnostics();
+
+    /**
+     *
+     * @return if the fixed file is different from the input file, this method return non empty unified diff
+     */
+    Optional<String> getUnifiedDiff();
+
+    ScalafixPatch[] getPatches();
+
+    /**
+     *
+     * @return It returns the new content file after scalafix has run on it if no error. The original file will stay unchanged.
+     */
+    Optional<String> previewPatches();
+
+    /**
+     *
+     * @return apply all patches and write the result to file.
+     */
+    ScalafixError[] applyPatches();
+
+    /**
+     * @param patches the patches should belong to this scalafixOutput's patches. If not, no patch will be applied to the file.
+     *
+     * @return It returns the new content file after applying the desired patches if no error.
+     * The original file will stay unchanged.
+     */
+    Optional<String> previewSelectivePatches(ScalafixPatch[] patches);
```

Comment:
can't this be an overload?
```suggestion
    Optional<String> previewPatches(ScalafixPatch[] patches);
```

Reply:
good idea. I forgot we can do this